src/components/travel-search-filter/utils.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  Mode,
+  TransportSubmode,
+} from '@atb/modules/graphql-types/journeyplanner-types_v3.generated';
+import {
+  TransportModeType,
+  TransportSubmodeType,
+} from '@atb/components/travel-search-filter/types';
+
+enum LegMode {
+  AIR = 'air',
+  BICYCLE = 'bicycle',
+  BUS = 'bus',
+  CABLEWAY = 'cableway',
+  CAR = 'car',
+  COACH = 'coach',
+  FOOT = 'foot',
+  FUNICULAR = 'funicular',
+  LIFT = 'lift',
+  METRO = 'metro',
+  RAIL = 'rail',
+  TRAM = 'tram',
+  WATER = 'water',
+}
+
+export type AnyMode = LegMode | Mode | TransportModeType | 'flex';
+export type AnySubMode = TransportSubmodeType;
+
+const TRANSPORT_SUB_MODES_BOAT: AnySubMode[] = [
```

Comment:
@mortennordseth, some of this might overlap with the implementation you did with the TransportMode component. How should we handle duplicates?

Reply:
Yup, there's some overlap. For example `TRANSPORT_SUB_MODES_BOAT` and `getTransportModeIconPath` already exist in `@atb/components/transport-mode/transport-icon/index.tsx`. I think we should coordinate these types and functions to avoid duplicates 🤔 Maybe we should add `@atb-as/config-specs` and use that as the standard?

Reply:
The transport mode types at least. If that makes them too connected we probably need duplicates, then? 🤔

Reply:
It might not. If we expand on the transport-mode to work with the travel search filter that would work. Then we can contain those types in the transport-mode folder and only use them in the travel search filter.